### PR TITLE
YLP-1841 Rework of accompanying documents in footer

### PR DESCRIPTION
**Pull request by fatb38**

### CODE REVIEW

**Comment by fatb38** [comment]
Don't forget to add a readme to explain how to get the html

File: packages/yourloops/components/footer/accompanying-document-links.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import React, { FunctionComponent } from 'react'
+import Link from '@material-ui/core/Link'
+import { Link as RouterLink } from 'react-router-dom'
+import Box from '@material-ui/core/Box'
+import { ClassNameMap } from '@material-ui/styles/withStyles'
+import { useTranslation } from 'react-i18next'
+import diabeloopUrls from '../../lib/diabeloop-url'
+import { useAuth } from '../../lib/auth'
+
+const AccompanyingDocumentLinks: FunctionComponent<{ classes: ClassNameMap }> = ({ classes }) => {
```
Comment (TimCanu) [inline packages/yourloops/components/footer/accompanying-document-links.tsx:37]: We usually extract common css add directly import it. I don't know what is better, just making sure that you remember.

File: packages/yourloops/pages/intented-use/intended-use-page.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import rawHtmlEN from './raw-html/EN'
+import { getCurrentLang } from '../../lib/language'
+import rawHtmlFR from './raw-html/FR'
+
+const IntendedUsePage: FunctionComponent = () => {
+  const getHtml = (): string => {
+    switch (getCurrentLang()) {
+      case 'fr':
+        return rawHtmlFR
+      default:
```
Comment (TimCanu) [inline packages/yourloops/pages/intented-use/intended-use-page.tsx:40]: Can you confirm that we only want to support french and english for now?

File: packages/yourloops/pages/intented-use/intended-use-page.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        return rawHtmlEN
+    }
+  }
+
+  const [html, setHtml] = useState<string>(getHtml())
+
+  i18n.on('languageChanged', () => {
```
Comment (TimCanu) [inline packages/yourloops/pages/intented-use/intended-use-page.tsx:47]: :bikini:

File: packages/yourloops/test/integration/pages/intended-use/intended-use-page.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import React from 'react'
+import userEvent from '@testing-library/user-event'
+import { act, render, screen } from '@testing-library/react'
+import { checkFooter } from '../../assert/footer'
+import i18n from 'i18next'
+
+describe('Intended use page', () => {
+  const history = createMemoryHistory({ initialEntries: ['/'] })
+
+  function getPage() {
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/intended-use/intended-use-page.spec.tsx:41]: There is a method `renderPage` in render.tsx. You don't need that

File: packages/yourloops/test/integration/pages/intended-use/intended-use-page.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      i18n.changeLanguage('fr')
+    })
+    expect(screen.getByText('Usage prévu et informations réglementaires')).toBeInTheDocument()
+    expect(screen.getByText('Fabricant')).toBeInTheDocument()
+  })
+
+  it('should render the yourloops version and latest release date on all languages', () => {
+    history.location.pathname = '/intended-use'
+    render(getPage())
+    expect(screen.getByText('YourLoops, version 1.0.0, released on 2000-01-01')).toBeInTheDocument()
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/intended-use/intended-use-page.spec.tsx:80]: This assertion fails you lazy bum

File: packages/yourloops/pages/intented-use/raw-html/EN.ts [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+    <p class="c22"><span class="c17 c33">WARNING</span><span class="c17">: </span><span class="c21">This software does not provide any medical advice and may not be relied upon for that purpose. </span><span class="c21">Home users</span><span class="c21">&nbsp;must consult a healthcare professional before making any medical interpretation and therapy adjustments from the information in the software.</span></p>
+    <p class="c19"><span class="c1"></span></p>
+    <p class="c22"><span class="c33 c17">WARNING</span><span class="c17">: </span><span class="c21">Healthcare professionals should use information in the software in conjunction with other clinical information available to them.</span></p>
+    <p class="c19"><span class="c4"></span></p>
+    <p class="c18"><span class="c10">Legal Manufacturer </span></p>
+    <hr>
+    <p class="c0"><span class="c2"></span></p>
+    <p class="c0"><span class="c2"></span></p><a id="t.01dd871b06081fb0ac6b9fb856ac0aca41b364bf"></a><a id="t.0"></a>
+    <table class="c24">
+      <tr class="c35">
```
Comment (TimCanu) [inline packages/yourloops/pages/intented-use/raw-html/EN.ts:489]: This shit code is making this warning:

`<tr> cannot appear as a child of <table>. Add a <tbody>, <thead> or <tfoot> to your code to match the DOM tree generated by the browser.`

I don't know if it makes sense to fix it

File: packages/yourloops/test/unit/components/footer/footer.test.tsx [hunk 7]
```diff
@@ -102,6 +106,25 @@ describe('Footer', () => {
     checkLinkHref('footer-link-url-privacy-policy', diabeloopUrls.getPrivacyPolicyUrL(i18n.language))
   })
 
+  it('should training link redirect to correct url', async () => {
+    await mountComponent()
+    checkLinkHref('footer-link-url-training', diabeloopUrls.getTrainingUrl(i18n.language))
+  })
+
+  it('should intended use link redirect to /intended-use page when user is logged in', async () => {
```
Comment (TimCanu) [inline packages/yourloops/test/unit/components/footer/footer.test.tsx:114]: The two following test can be removed as the link redirect is covered by an IT

**Review by TimCanu — COMMENTED** [review]
Your images could use a better naming

File: packages/yourloops/pages/intented-use/intended-use-page.tsx (hunk 2 above)
Comment (fatb38) [reply packages/yourloops/pages/intented-use/intended-use-page.tsx:40]: For now it's the plan. Futur will tell more later !

File: packages/yourloops/app/main-lobby.tsx [hunk 8]
```diff
@@ -40,12 +40,16 @@ import PatientConsentPage from '../pages/patient/patient-consent'
...
 
 const RENEW_CONSENT_PATH = '/renew-consent'
 const NEW_CONSENT_PATH = '/new-consent'
 const COMPLETE_SIGNUP_PATH = '/complete-signup'
-const PUBLIC_ROUTES = ['/login']
-const EXTERNAL_THEME_ROUTES = [NEW_CONSENT_PATH, RENEW_CONSENT_PATH, COMPLETE_SIGNUP_PATH, ...PUBLIC_ROUTES]
+const LOGIN_PATH = '/login'
+const INTENDED_USE_PATH = '/intended-use'
+const PUBLIC_ROUTES = [LOGIN_PATH]
+const ALL_ACCESS_ROUTES = [INTENDED_USE_PATH]
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/app/main-lobby.tsx:51]: This comment has not been fixed

**Review by TimCanu — COMMENTED** [review]
First comment made was not fixed

**Review by TimCanu — APPROVED** [review]
Looking good :+1: 
Only merge if the BETA has been run on yourlooops CI.

**Pull request merged**